### PR TITLE
Feature/ Setup and initialize wagmi

### LINKED ISSUE #349: [Feature] Move from EtherJs to viem, by using wagmi
## Description

We need to move from etherjs to Viem, and use wagmi on top of it. 

## Acceptance Criteria

- [x] Wagmi setup and first use.
- [ ] Refactoring all existing feature with Wagmi

### PR DESCRIPTION
**Pull request by dasarathg68**
Fixes #349

## Type of change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

* **Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines

### CODE REVIEW

**Review by hermannleboss — APPROVED** [review]
Look Good to me

**Comment by dasarathg68** [comment]
LGTM.

**Comment by dasarathg68** [comment]
**Notes for future reference:**
#350 sets up and initializes wagmi in the frontend. 
Further, it uses a function setupApp() to improve testability of the main.ts file.
```typescript
export function setupApp() {
  const app = createApp(App)
  const queryClient = new QueryClient()

  app.use(createPinia())
  app.use(router)
  app.use(WagmiPlugin, { config })
  app.use(VueQueryPlugin, { queryClient })

  return app
}
setupApp().mount('#app')
```

I have set wagmi.config.ts based on best practices that allow us to leverage both the @wagmi/core and @wagmi/vue libraries since some composables like deploy aren't available for Vue just yet.
In a different project, I have played around with all of the functions that we would need for the CNC portal and if you need a reference, then here's what you would need to use:
https://github.com/dasarathg68/GovernanceX/blob/main/app/src/views/HomeView.vue
https://wagmi.sh/vue/getting-started
https://wagmi.sh/core/getting-started

**Comment by dasarathg68** [comment]
**Notes for future reference:**
https://github.com/globe-and-citizen/cnc-portal/issues/349#issuecomment-2352224975

File: app/src/main.ts [hunk 1]
```diff
@@ -1,14 +1,22 @@
...
+import { config } from './wagmi.config'
+import { QueryClient, VueQueryPlugin } from '@tanstack/vue-query'
 import { createApp } from 'vue'
 import { createPinia } from 'pinia'
-
 import App from './App.vue'
 import router from './router'
 
-const app = createApp(App)
+export function setupApp() {
```
Comment (hermannleboss) [inline app/src/main.ts:10]: Do you need to export setup app here ?

Comment (hermannleboss) [reply app/src/main.ts:10]: @dasarathg68

Comment (dasarathg68) [reply app/src/main.ts:10]: I can't test main.ts directly so I have created a function setupApp that uses WagmiPlugin, router, pinia, VueQueryPlugin and exported it. 
Below I have mounted it by calling the setupApp() function. 
Like this:
```typescript
setupApp().mount('#app')
```

**Pull request merged**